pkg/model/openstackmodel/network.go

```diff
@@ -50,6 +50,10 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 	}
 
 	needRouter := true
+	//Do not need router if there is no external network
+	if b.Cluster.Spec.CloudConfig.Openstack.Router.ExternalNetwork == nil {
```

Comment:
I think you'll want to do additional checking here to avoid a nil pointer dereference

Reply:
Yep. Thanks for catching.